src/git.ts

```diff
@@ -0,0 +1,72 @@
+import fs from "fs";
+import path from "path";
+
+import simpleGit, { Response } from "simple-git";
+import tmp from "tmp";
+
+/**
+ * Apply the input Git patches and push the changes.
+ * @param {string} remoteURL the Git URL to clone from and push to. It should include any required authentication.
+ * @param {string} sourceBranch the branch to use as a base for applying the patches.
+ * @param {string} targetBranch the branch to push applied patches to.
+ * @param {Array<string>} patches the array of Git patch strings to apply.
+ * @return {Promise<void>} a promise that resolves to nothing and rejects when there is a Git error such as a patch
+ *   not applying.
+ */
+export async function applyPatches(
+  remoteURL: string,
+  sourceBranch: string,
+  targetBranch: string,
+  patches: Array<string>,
+): Promise<void> {
+  return new Promise((resolve, reject) => {
+    const dirObj = tmp.dirSync({ keep: true, unsafeCleanup: true });
+    const patchFiles: Array<string> = [];
+
+    try {
+      for (let i = 0; i < patches.length; i++) {
+        if (!patches[i]) {
+          continue;
+        }
+
+        const patchPath = path.join(dirObj.name, `patch${i + 1}.patch`);
+        fs.writeFileSync(patchPath, patches[i]);
+        patchFiles.push(patchPath);
+      }
```

Comment:
Not necessary, but I feel like it would be more JS idiomatic to use `forEach` or `map` here. Granted, I'm not a JS expert either, and I'm not 100% sure how it would behave inside the `try`.

Eg
```
patchFiles = patches.map((patch, i) => {
  const patchPath = path.join(dirObj.name, `patch${i + 1}.patch`);
  fs.writeFileSync(patchPath, patch);
  return patchPath
})
```